.github/workflows/dependency-review-v2.yml

```diff
@@ -0,0 +1,127 @@
+        description: |
+          The type of machine to run the job on. Must be provided as a stringified list (e.g. public repos should specify `runs-on: '["ubuntu-latest"]'`)
+        default: '["coveo", "arm64" , "linux", "eks"]'
+        type: string
+      retry-on-snapshot-warnings:
+        description: Whether to retry on snapshot warnings (to be used for projects where a dependency submission Action is used)
+        required: false
+        type: boolean
+        default: false
+      retry-on-snapshot-warnings-timeout:
+        description: Number of seconds to wait before stopping snapshot retries.
+        required: false
+        type: number
+        default: 120
+      warn-on-openssf-scorecard-level:
+        description: Numeric threshold for the OpenSSF Scorecard score. If the score is below this threshold, the action will warn you.
+        required: false
+        type: number
+        default: 3
+
+permissions: { }
+
+jobs:
+  dependency-review:
+    name: Dependency Review
+    runs-on: ${{ fromJson(inputs.runs-on) }}
+
+    permissions:
+      contents: read
+      pull-requests: write
+
+    steps:
+      - name: Checkout scan target
+        uses: actions/checkout@v4
+
+      - name: Checkout licenses
+        uses: actions/checkout@v4
+        with:
+          repository: coveo/dependency-allowed-licenses
+          path: coveo-dependency-allowed-licenses
+
+      - name: Get Properties
+        uses: actions/github-script@v7
+        id: get-properties
+        with:
+          script: |
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
+
+            const repoDetails = await github.request('GET /repos/' + owner + '/' + repo, {
+              owner: owner,
+              repo: repo
+            });
+            const isPublic = !repoDetails.data.private;
+            console.log(`Is this a public repo? ${isPublic}`);
+            core.setOutput('is_public', isPublic);
+
+            let distributedValue = isPublic;
+            if (!isPublic) {
+              const response = await github.request('GET /repos/' + owner + '/' + repo + '/properties/values', {
+                owner: owner,
+                repo: repo
+              });
+              console.log('Repository properties: ' + JSON.stringify(response.data, null, 2));
+              const distributedProperty = response.data.find(prop => prop.property_name === 'is_distributed');
+              distributedValue = distributedProperty ? distributedProperty.value === 'true' : true;
+            }
+            core.setOutput('is_distributed', distributedValue);
+
+      - name: Select configuration
+        id: select-config
+        uses: actions/github-script@v7
+        with:
+          result-encoding: string
+          script: |
+            const isPublic = ${{ steps.get-properties.outputs.is_public }} === 'true';
+            const isDistributed = ${{ steps.get-properties.outputs.is_distributed }} === 'true';
```

Comment:
Typically, we use `dash-case` in GitHub for pretty much everything. Not saying you should but that's what I see most of the time.

Reply:
I used underscore case to match the "property" underscore case.

The property is underscore-case because properties previously created by the DevPortal team are also underscore-case:

<img width="981" alt="Screenshot 2024-11-13 at 8 05 20 AM" src="https://github.com/user-attachments/assets/07b5298d-5541-4f19-a7bc-7b929d05fe6e">

So, I can move the output to dash-case, but leave the property in underscore-case. Is that what you would do?

Reply:
Thanks for the explanation! I think they should match 1:1, so let's leave it like this.